proto/private/protoc/prebuilt_protoc_toolchain.bzl

```diff
@@ -0,0 +1,51 @@
+"""Repository rule that relies on fetching a protoc binary from GitHub Releases.
+
+It exposes bin/protoc as a proto_toolchain.
+"""
+
+load(":versions.bzl", "PROTOC_PLATFORMS", "PROTOC_VERSIONS")
+
+def _prebuilt_protoc_repo_impl(rctx):
+    release_version = rctx.attr.version
+
+    filename = "{}-{}-{}.zip".format(
+        "protoc",
+        release_version.removeprefix("v"),
+        rctx.attr.platform,
+    )
+    url = "https://github.com/protocolbuffers/protobuf/releases/download/{}/{}".format(
+        release_version,
+        filename,
+    )
+    rctx.download_and_extract(
+        url = url,
+        integrity = PROTOC_VERSIONS[release_version][filename],
+    )
+    rctx.file("BUILD.bazel", """\
+# Generated by @rules_proto//proto/toolchains:prebuilt_protoc_toolchain.bzl
+load("@rules_proto//proto/private/rules:proto_toolchain_rule.bzl", "proto_toolchain")
+
+proto_toolchain(
+    name = "prebuilt_protoc_toolchain",
+    proto_compiler = "{protoc_label}",
+    visibility = ["//visibility:public"],
+)
+""".format(
+        protoc_label = ":bin/protoc.exe" if rctx.attr.platform.find("windows") != -1 else ":bin/protoc",
```

Comment:
The key for Windows is `win64`. Alternatively, you could use `rctx path("bin/protoc.exe").exists".